dallinger/frontend/static/scripts/dallinger.js

```diff
@@ -13,6 +13,31 @@ var getUrlParameter = function getUrlParameter(sParam) {
         }
     }
 };
+
+var Dallinger = (function () {
+  var dlgr = {};
+
+  participantId = getUrlParameter("participant_id") || -1;
+
+  dlgr.submitQuestionnaire = function (name, callback) {
+    formSerialized = $("form").serializeArray();
+    formDict = formSerialized.reduce(
+      function(m,o){ m[o.name] = o.value; return m; }, {}
+    );
+    $.ajax({
+      type: "POST",
+      url: "/question/" + participantId,
+      data: {
+        question: name || "questionnaire",
+        number: 1,
+        response: JSON.stringify(formDict),
```

Comment:
I agree it will be good to change the Question table so that it has a JSONB column, as you say. I'd also like to some day remove all the `property_x` columns and replace them with a single JSONB column. However, these are breaking changes and it'd be better to do them all at once, perhaps as part of a 4.0 release. In the meantime, this gets around the issue of needing to submit 1 HTTP request per question with minimal fuss. Does this seem like an okay (temporary) workaround?